runtime/src/main/java/org/corfudb/protocols/wireprotocol/ICorfuPayload.java

```diff
@@ -271,7 +294,7 @@
      * @param <T>           The type of the payload.
      * */
     @SuppressWarnings("unchecked")
-    static <T> void serialize(ByteBuf buffer, T payload) {
+    static <T> void serialize(ByteBuf buffer, @Nonnull T payload) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)